sklearn/tree/_tree.pyx

```diff
@@ -391,29 +548,49 @@ cdef class Entropy(ClassificationCriterion):
 
         cdef double entropy = 0.0
         cdef double total = 0.0
-        cdef double tmp
+        cdef double pmk
         cdef SIZE_t k
         cdef SIZE_t c
 
+        # Go through each response and calculate the entropy
         for k in range(n_outputs):
+            # Initialize the entropy to 0.0 for this response
             entropy = 0.0
 
+            # For each class in this response
             for c in range(n_classes[k]):
-                tmp = label_count_total[c]
-                if tmp > 0.0:
-                    tmp /= weighted_n_node_samples
-                    entropy -= tmp * log(tmp)
+                # Look up the weighted number of points in this class
+                pmk = label_count_total[c]
+
+                # If more than 0 weight, then calculate entropy
+                if pmk > 0.0:
+                    pmk /= weighted_n_node_samples
+                    entropy -= pmk * log(pmk)
 
+                # Otherwise the addition is 0.
+            # Add that class entropy to the total entropy
             total += entropy
             label_count_total += label_count_stride
 
+        # Return the total entropy over all responses divided by the number of
+        # responses.
```

Comment:
+1, this code is quite straight forward. You might want to rename `total` to `total_entropy` if you want to make it fully explicit. No need for inline comments in this function.